Northstar.CustomServers/mod/scripts/vscripts/weapons/sh_phase_shift.gnut

```diff
@@ -0,0 +1,774 @@
+	PrecacheParticleSystem( SHIFTER_APPEAR_FX_TITAN )
+	PrecacheParticleSystem( SHIFTER_DISAPPEAR_FX )
+	PrecacheParticleSystem( SHIFTER_DISAPPEAR_FX_TITAN )
+	PrecacheParticleSystem( SHIFTER_PERSONAL_FX )
+	PrecacheParticleSystem( SHIFTER_SCREEN_FX )
+	PrecacheParticleSystem( SHIFTER_SCREEN_FX_START )
+
+	MpAbilityShifterWeapon_Init()
+
+	RegisterSignal( "StartPhaseShift" )
+	RegisterSignal( "StopPhaseShift" )
+	RegisterSignal( "ForceStopPhaseShift" )
+}
+
+int function PhaseShift( entity ent, float warmupTime, float duration )
+{
+	if ( !IsAlive( ent ) )
+		return 0
+
+	if ( ent.IsPhaseShifted() )
+		return 0
+
+	// PROTO: this should be better eventually
+	// this is to prevent weirdness if you're already in a context action
+	if ( ent.IsPlayer() )
+	{
+		entity proxy = ent.GetFirstPersonProxy()
+		if ( ( ent.ContextAction_IsActive() || (proxy != null && proxy.Anim_IsActive()) )
+			&& !ent.IsZiplining()
+			&& !ent.ContextAction_IsLeeching()
+			&& ent.GetTitanSoulBeingRodeoed() == null )
+		{
+			return 0
+		}
+
+		ent.PhaseShiftBegin( warmupTime, duration )
+
+		#if SERVER
+			thread ServerPhaseShiftPlayerThread( ent, warmupTime, duration )
+		#else
+			thread ClientPhaseShiftWarmupThread( ent, warmupTime )
+		#endif // #if SERVER
+
+		return 1
+	}
+	else
+	{
+		ent.PhaseShiftBegin( warmupTime, duration )
+
+		#if SERVER
+			thread ServerPhaseShiftPlayerThread( ent, warmupTime, duration )
+		#endif // #if SERVER
+
+		return 1
+	}
+
+	unreachable
+}
+
+#if SERVER
+void function ServerPhaseShiftPlayerThread( entity ent, float warmupTime, float shiftTime )
+{
+	ent.EndSignal( "OnDeath" )
+	ent.EndSignal( "ForceStopPhaseShift" )
+
+	// initialise the two ent.s variables if needed
+	if ( !( "oldPhaseStateLock" in ent.s ) || !( "oldPhaseState" in ent.s ) )
+	{
+		ent.s.oldPhaseStateLock <- false
+		ent.s.oldPhaseState <- false
+	}		
+
+	bool keepAnimActive = false
+
+	if ( ent.IsPlayer() )
+	{
+		ent.EndSignal( "player_embarks_titan" )
+
+		Leech_Disallow( ent )
+	//	EmitSoundOnEntityExceptToPlayer( ent, ent, SHIFTER_PRE_SOUND_3P )
+
+		entity proxy = ent.GetFirstPersonProxy()
+		if ( proxy != null )
+			proxy.SetForceVisibleInPhaseShift( true )
+
+		if ( Rodeo_IsAttached( ent ) )
+		{
+			if ( ent.GetParent() != null && ent.GetParent().IsPhaseShifted() )
+				keepAnimActive = true
+			else
+				ent.Signal( "RodeoOver" )
+		}
+
+		if ( ent.IsTitan() )
+		{
+			UntrackAllToneMarks( ent )
+			entity titanSoul = ent.GetTitanSoul()
+			if ( titanSoul != null )
+			{
+				if ( titanSoul.soul.batteryContainer != null )
+				{
+					titanSoul.soul.batteryContainer.SetForceVisibleInPhaseShift( true )
+				}
+			}
+		}
+
+		Battery_StopFXAndHideIconForPlayer( ent )
+	}
+	else
+	{
+		if ( ent.IsTitan() )
+			UntrackAllToneMarks( ent )
+		ent.EnableNPCFlag( NPC_IGNORE_ALL )
+		EmitSoundOnEntity( ent, SHIFTER_PRE_SOUND_3P )
+	}
+
+	// only change the oldPhaseState if it's not locked
+	if ( !ent.s.oldPhaseStateLock )

```

Comment:
```suggestion
	if ( !ent.s.oldPhaseStateLock || !GetConVarBool("ns_use_phase_fix") )
```
I can't suggest changes that arent near to changes you made, but this + adding the ConVar to the mod.json with a default value of 1 should do what Bob said i think